docs/creating_integration_manifest.md

```diff
@@ -179,6 +179,41 @@ MQTT discovery works by subscribing to MQTT topics specified in the manifest.jso
 }
 ```
 
+## DHCP
+
+If your integration supports discovery via dhcp, you can add the type to your manifest. If the user has the `dhcp` integration loaded, it will load the `dhcp` step of your integration's config flow when it is discovered. We support passively listening for DHCP discovery by the `hostname` and [OUI](https://en.wikipedia.org/wiki/Organizationally_unique_identifier). The manifest value is a list of matcher dictionaries, your integration is discovered if all items of any of the specified matchers are found in the DHCP data. It's up to your config flow to filter out duplicates.
+
+If the integration supports `zeroconf` or `ssdp`, these should be preferred over `dhcp` as it generally offers a better
+user experience.
+
+The following example has three matchers consisting of two items. All of the items in any of the three matchers must match for discovery to happen by this config.
+
+For example:
+
+-  If the `hostname` was `Rachio-XYZ` and the `macaddress` was `00:9D:6B:55:12:AA`, the discovery would happen.
+-  If the `hostname` was `Rachio-XYZ` and the `macaddress` was `00:00:00:55:12:AA`, the discovery would not happen.
```

Comment:
Great, looks much cleaner now! One more nitpick, assuming the first entry matches, I think it's worth mentioning that the hostname checks are case-insensitive.